docs/DevEnvironment.md

```diff
@@ -0,0 +1,158 @@
+<!--
+# @title Development Environment
+-->
+# Development Environment
+
+## Introduction
+
+Rouge is written in Ruby and has a number of development dependencies. To
+develop new features for Rouge (like a lexer for a new syntax) or to fix bugs,
+you need to set up a development environment.
+
+### Ruby and Git
+
+First thing's first. This guide is _not_ a guide to installing Ruby and Git.
+There are a number of excellent resources out there explaining how to do these
+things. For Ruby, we recommend [the official documentation][rb-inst-docs] and
+for Git, [GitHub's documentation][gh-inst-docs].
+
+[rb-inst-docs]: https://www.ruby-lang.org/en/documentation/installation/
+[gh-inst-docs]: https://help.github.com/en/articles/set-up-git
+
+### Terminal Emulators
+
+This guide assumes you are familiar with the _command line_. The command line is
+accessed through a _terminal emulator_. In macOS, the default emulator is called
+"Terminal" and can be found by searching for "Terminal" in Spotlight. In
+Windows, depending on your version, you may want to install a third-party
+emulator like [Cygwin][]. In Linux, well, you're probably reading this at the
+command line.
+
+[Cygwin]: https://cygwin.com/
+
+### GitHub
+
+The official Rouge repository is on GitHub and we use GitHub to coordinate
+development. While you don't _need_ a GitHub account to hack on Rouge, you're
+going to need one to contribute your improvements back to the Rouge community.
+Creating [an account][gh-make-acc] is free of charge. And just think how awesome
+it'll be when you take your collaboration to the "next level".
+
+[gh-make-acc]: https://github.com/join
+
+## Getting Rouge
+
+### Forking the Repository
+
+To develop Rouge, we're going to create a "fork" of Rouge where we can make our
+changes. These will be happily isolated from everyone else and then—through the
+magic of Git—mergeable back into the main project later.
+
+First, visit [the front page][rouge-fp] of Rouge's repo on GitHub. Unless you're
+crazy and are trying to develop code on your phone (like perhaps I am doing
+right now), you'll see a button near the top right of your browser window that
+will say "Fork". Click this and GitHub will ask where you want to create your
+fork. Select your account and—boom!—you've just forked Rouge.
+
+[rouge-fp]: https://github.com/rouge-ruby/rouge
+
+### Cloning Your Fork
+
+The next thing to do is to get your fork onto your computer. Git makes this
+easy. In the directory you want to hold your repository, type:
+
+```shell
+git clone git@github.com:<your_github_account_name>/rouge.git
+```
+
```

Comment:
Instead of typing this, it'd be easier for the novice to click on the **`Clone or Download`** button, copy the URI and paste that in the terminal:
```
git clone https://github.com/<username>/rouge.git
```

Reply:
Yes, though I do like how it helps folks make sure they're not accidentally cloning the original repo.

Reply:
That reminds me.. Users need to know about tracking the upstream (original) repository to pull in the latest `master` for a subsequent PR.
You need not write those instructions here. Just including the relevant link(s) to GitHub Help docs will complete this guide.

Reply:
I thought about suggesting they click or download but I want to leave it as is so that they don't accidentally clone the official repo.

As for tracking upstream, will add that in.